RevenueCatUI/Templates/V2/ViewHelpers/BadgeModifier.swift

```diff
@@ -0,0 +1,468 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  BadgeModifier.swift
+//
+//  Created by Mark Villacampa 09/12/2024.
+
+// swiftlint:disable file_length
+
+import RevenueCat
+import SwiftUI
+
+#if PAYWALL_COMPONENTS
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+struct BadgeModifier: ViewModifier {
+
+    let badge: BadgeInfo?
+
+    struct BadgeInfo {
+        let style: PaywallComponent.BadgeStyle
+        let alignment: PaywallComponent.TwoDimensionAlignment
+        let stack: PaywallComponent.CodableBox<PaywallComponent.StackComponent>
+        let badgeViewModels: [PaywallComponentViewModel]
+        let stackShape: ShapeModifier.Shape?
+        let uiConfigProvider: UIConfigProvider?
+    }
+
+    func body(content: Content) -> some View {
+        if let badge = badge {
+            content.apply(badge: badge)
+        } else {
+            content
+        }
+    }
+}
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+fileprivate extension View {
+
+    @ViewBuilder
+    func apply(badge: BadgeModifier.BadgeInfo) -> some View {
+        switch badge.style {
+        case .edgeToEdge:
+            self.applyBadgeEdgeToEdge(badge: badge)
+        case .overlaid:
+            self.overlay(
+                VStack(alignment: .leading) {
+                    VStack {
+                        ComponentsView(componentViewModels: badge.badgeViewModels, onDismiss: {})
```

Comment:
I'm not entirely sure we need to pass the `onDismiss` block here? I assume it is only used for the NavigateBack action button and would only make sense if we want to include buttons in the children elements here?

Reply:
Yeah yeah, this if fine 😅